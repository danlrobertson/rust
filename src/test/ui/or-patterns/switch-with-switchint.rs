// run-pass
#![feature(or_patterns)]
//~^ WARN the feature `or_patterns` is incomplete and may cause the compiler to crash

#[derive(Debug)]
enum Foo {
    Bar(usize),
    Baz(usize, usize),
}

fn test(x: Option<Foo>) -> bool {
    match x {
        Some(Foo::Bar(1024) | Foo::Baz(2048, 4096)) => true,
        Some(_) => false,
        None => false,
    }
}

fn main() {
    assert!(test(Some(Foo::Bar(1024))));
    assert!(!test(Some(Foo::Bar(1025))));
    assert!(test(Some(Foo::Baz(2048, 4096))));
    assert!(!test(Some(Foo::Baz(2048, 4097))));
    assert!(!test(Some(Foo::Baz(2049, 4096))));
}
