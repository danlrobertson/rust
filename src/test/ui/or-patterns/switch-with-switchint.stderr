warning: the feature `or_patterns` is incomplete and may cause the compiler to crash
  --> $DIR/switch-with-switchint.rs:2:12
   |
LL | #![feature(or_patterns)]
   |            ^^^^^^^^^^^

